docs/DeveloperGuide.md

```diff
@@ -1,13 +1,304 @@
+
+After loading existing saved cashflows from data.txt, the program will check for cashflows that are set to be recurring and has not recurred.
+
+Example:
+```
+if (recur > 0 && !hasRecurred) {
+    ...            
+}
+```
+
+#### Step 3
+Once a cashflow matches the above criteria, the date of its next addition to the Financial Planner, `dateOfAddition`, would be determined.
+
+`dateOfAddition` would be compared to the current date, and if the current date is after or equal to `dateOfAddition`, an identical cashflow would be instantiated.
+
+This identical cashflow would then have its `date` variable set to `dateOfAddition`, then this cashflow would be added to a temporary list, `tempCashflowList`.
+
+The original cashflow would then have its `hasRecurred` variable be set to `true`.
+
+#### Step 4
+Each cashflow in `tempCashflowList` goes through **Step 3** again, so that multiple cashflows can be added if it has recurred more than once.
+
+Once the process is done, all cashflows in `tempCashflowList` are then added to the Financial Planner.
+
+The added cashflows are then displayed to the user.
+
+#### Recurring Cashflow Class Diagram
+Given below is the class diagram showing the class structure of the recurring cashflow mechanism:
+![](images/cashflow/RecurClassDiagram.png)
+#### Recurring Cashflow Sequence Diagrams
+Given below is the sequence diagram showing the recurring cashflow mechanism:
+![](images/cashflow/RecurSequence.png)
+![](images/cashflow/AddRecurringSequence.png)
+
+### Budget Feature
+
+This feature has 5 functions, `set`, `update`, `delete`, `reset`, and `view`.
+
+![](images/Budget.png)
+
+The BudgetCommand will execute the appropriate command and print through `Budget.java` and prints any message to the user through `Ui.java`.
+
+#### Set and update budget:
+
+Example:
+```
+budget set /b 500
+budget update /b 1000
+```
+The '/b' is followed by the budget amount. The first line will set the budget by calling `setBudget(500)` method in `Budget.java`.
+The second line updates the budget by adding or subtracting the difference between the initial and updated amount to the 
+initial and current budget. This is done through `updateBudget(500)` method in `Budget.java`. Both functions can be seen 
+in the diagram above
+
+#### Delete budget:
+
+![](images/deleteBudget.png)
+
+The budget will be deleted by setting the initial and current budget to 0 through the `deleteBudget()` method in `Budget.java`.
+
+Example: `budget delete`
+
+#### Reset budget:
+
+![](images/resetBudget.png)
+
+The budget will be reset by resetting the current budget to the initial budget through the `resetBudget()` method in `Budget.java`.
+
+Example: `budget reset`
+
+#### View budget:
+
+![](images/viewBudget.png)
+
+The current budget will be shown to the user through the `Ui`.
+
+Example: `budget view`
+
 
 ## Product scope
 ### Target user profile
```

Comment:
Perhaps you could have a clear target user profile which could give the reader a better picture of who the code should cater for